src/clients/identity/types.ts

```diff
@@ -11,6 +11,7 @@ export interface AuthHeader {
 export interface IValidateTokenOptions {
   roles?: string[];
   permissions?: string[];
+  withRolesAndPermissions?: boolean;
```

Comment:
why you need this?

Reply:
because some tokens don't contain permissions in the JWT payload; without permissions the entitlements check cannot operate properly

the intention of that flag was to force the identity client to ensure that permissions list is present in decoded token

Reply:
maybe worth renaming the property?
enforcePermissionsOnToken/enforceAccessControl or something that tells us it enforces something on the token validation.
It's a bit confusing since roles and permissions can be passed separately and withRolesAndPermissions makes me think like I must also pass roles and permissions